wordpress-mobile/WordPress-iOS
# Gutenberg: Don't save empty posts

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/369

While creating this PR I noticed the following:
 - Gutenberg JS  doesn't send auto-save messages when  the post is empty, this can cause some sync problems with the status of the Publish button ( being active when it shouldn't). **Update:** I proposed a fix for this issue in this GB PR: https://github.com/WordPress/gutenberg/pull/23233, if you want to test with this active start Metro using this branch: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2393
 - I also found some issue where the message sent from GB, has the variable `hasChanges` set to false when there is changes, this is also confusing the status of the Navigation Publish/Update button, thinking that the post has no changes when it has changes.
 - To make sure the post is never published empty I add a failsafe before the Publish action happens.

To test:
 - Create a new post/page on the editor
 - Write some content (title or post content)
 - Check that the publish button is enabled
 - Delete all content from title and content
 - If the Publish button is enabled, press Publish
 - Check that an alert shows up explaining that an empty post/page cannot be published
 - Now add a single image block and don't write any text.
 - Check that you can publish

PR submission checklist:
